### DISCUSSION TITLE
Why code generator / build runner?

### DISCUSSION THREAD

**Discussion by ghost** [isar/isar#923] [Ideas]
Awesome library. Local storage is really important. ... But I'm just wondering if we can do such great  stuff without manual code generation? I don't want to care about and don't like generated files in my source code. Ok, I have not enough experience in Flutter / Dart (coming from Angular / TypeScript). But I believe it must be possible without all that part.g.dart stuff. Isar could do this is the background in a separated generator path to leave a clean code. Btw. can someone please explain why this is currently needed? Or why not without. Or what needs to be done to go without that.

**Comment by simc** [comment]
Code generation is a very common pattern in Dart/Flutter. The Dart team is working on static meta programming which will remove the need for source file generation but until then there's nothing we can do. You can check out the files to determine what gets generated exactly :)
